src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs

```diff
@@ -182,16 +182,38 @@ private bool IsCompatibleWithTarget(NuGetFramework target, NuGetFramework candid
 
         private static bool IsCompatibleWithTargetCore(NuGetFramework target, NuGetFramework candidate)
         {
-            bool result = NuGetFramework.FrameworkNameComparer.Equals(target, candidate)
-                    && IsVersionCompatible(target.Version, candidate.Version)
-                    && StringComparer.OrdinalIgnoreCase.Equals(target.Profile, candidate.Profile);
-
-            if (target.IsNet5Era && candidate.HasPlatform)
+            bool result = true;
+            bool isNet6Era = StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, target.Framework) && target.Version.Major >= 6;
```

Comment:
I thought about it, but the reason we did this for net5 was because we had to check for it in so many places. The `isNet6Era` stuff doesn't seem significant enough to warrant that.

Reply:
My suggestion is not about the number of different places in the code we'd do it, but rather about the number of times we'd check it.

Reply:
To avoid allocations, you can do `target.IsNet5Era && target.Version.Major >= 6`.